lib/promscrape/discovery/ovhcloud/api.go

```diff
@@ -0,0 +1,76 @@
+package ovhcloud
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+)
+
+// mapping for endpoint names to their URI for external configuration
+var availableEndpoints = map[string]string{
+	"ovh-eu":        "https://eu.api.ovh.com/1.0",
+	"ovh-ca":        "https://ca.api.ovh.com/1.0",
+	"ovh-us":        "https://api.us.ovhcloud.com/1.0",
+	"kimsufi-eu":    "https://eu.api.kimsufi.com/1.0",
+	"kimsufi-ca":    "https://ca.api.kimsufi.com/1.0",
+	"soyoustart-eu": "https://eu.api.soyoustart.com/1.0",
+	"soyoustart-ca": "https://ca.api.soyoustart.com/1.0",
+}
+
+var configMap = discoveryutils.NewConfigMap()
+
+type apiConfig struct {
+	client *discoveryutils.Client
+
+	applicationKey    string `yaml:"application_key"`
+	applicationSecret string `yaml:"application_secret"`
```

Comment:
It's better to replace secret values string with `promauth.Secret`. It ll hide secret content at `/config` page.

Reply:
AFAIK fields of `apiConfig` struct should be used internally only:
https://github.com/jiekun/VictoriaMetrics/blob/c2c09c14424ef17ea61d542fcf208e7f51764681/lib/promscrape/discovery/ovhcloud/ovhcloud.go#L21-L22

And secret field for SDConfig is already defined as `*promauth.Secret`:
https://github.com/jiekun/VictoriaMetrics/blob/c2c09c14424ef17ea61d542fcf208e7f51764681/lib/promscrape/discovery/ovhcloud/ovhcloud.go#L22

Do you think we still need `promauth.Secret` for the `apiConfig`?